pkg/controllers/common/ingress_config.go

```diff
@@ -72,3 +78,42 @@ func GetCustomRouteHostname(ingress *configv1.Ingress, namespace string, name st
 	}
 	return ""
 }
+
+func UpdateComponentRouteStatus(ctx context.Context, client configsetterv1.IngressInterface, componentRoute *configv1.ComponentRouteStatus) (bool, error) {
+	// Override the timestamps
+	now := metav1.Now()
+	setLastTransactionTime(componentRoute, now)
+
+	updated := false
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		ingressConfig, err := client.Get(ctx, "cluster", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		existingComponentRoute := GetComponentRouteStatus(ingressConfig, componentRoute.Namespace, componentRoute.Name)
+		if existingComponentRoute != nil {
+			// Ignore timestamps on the existing route
+			setLastTransactionTime(existingComponentRoute, now)
```

Comment:
you shouldn't change the input, at least make a copy

Reply:
Done.